yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -113,7 +121,9 @@ export class Archiver implements L2BlockSource, L2LogsSource, ContractDataSource
     });
     const searchStartBlock = Number((await registryContract.read.getCurrentSnapshot()).blockNumber);
 
-    const archiverStore = new MemoryArchiverStore(config.maxLogs ?? 1000);
+    const archiverStore = rootDB
+      ? new LMDBArchiverStore(rootDB, config.maxLogs ?? 1000)
+      : new MemoryArchiverStore(config.maxLogs ?? 1000);
```

Comment:
I've left the old MemoryArchiverStore in as a reference for the code review and in case we might need it for other tests but otherwise all code paths should pass a `rootDB` now and use a temporary LMDB instance if persistance isn't needed.